src/api/getArtifactsForBranchAndWorkflow.ts

```diff
@@ -51,63 +47,37 @@ export async function getArtifactsForBranchAndWorkflow(
     }`
   );
 
-  let currentPage = 0;
-  let completedWorkflowRuns: WorkflowRun[] = [];
-
-  for await (const response of octokit.paginate.iterator(
-    octokit.rest.actions.listWorkflowRuns,
-    {
-      owner,
-      repo,
-      // Below is typed incorrectly, it needs to be a string but typed as number
-      workflow_id: (workflow_id as unknown) as number,
-      branch,
-      status: 'success',
-      per_page: PER_PAGE_LIMIT,
-    }
-  )) {
-    if (!response.data.length) {
-      core.warning(`Workflow ${workflow_id} not found in branch ${branch}`);
-      core.endGroup();
-      return null;
-    }
-
-    // XXX: This function is only used by `retrieveBaseSnapshots`, which means that base snapshots
-    // should only be used from the workflow org/repo and not from forks, as they can create
-    // a pull request that uses the base branch name.
-    //
-    // If this needs to be more generic, this should be an option.
-    const workflowRuns = response.data.filter(
-      workflowRun =>
-        workflowRun.head_repository.full_name === `${owner}/${repo}`
+  const {
+    data: {workflow_runs: workflowRuns},
+  } = await octokit.rest.actions.listWorkflowRuns({
+    owner,
+    repo,
+    // Below is typed incorrectly, it needs to be a string but typed as number
+    workflow_id,
+    branch,
+    status: 'success',
+    head_sha: commit,
```

Comment:
Nice work @billyvg 🙏🏼 So if we do this via the API, do we solve the bug that @narsaynorath reported or do we still have the problem of not finding artifacts for the run?

Reply:
So not exactly -- the problem @narsaynorath reported is due to the workflow failing and we can't really do much about that. While I tried to get it to re-run and pass though, I noticed another potential problem where we would need to paginate through a lot of results in order to find the correct workflow run (due to the amount of workflow runs we get these days).

Reply:
Nice, I suspect this will also make the actions faster as we'll possibly run less requests in order to find the right run 👏🏼